### PR TITLE
Introduce TableUpdateImpl copy that takes an MCS

### PR DESCRIPTION
**Pull request by devinrsmith**
This allows allows the majority of copy callsites to be more efficient when they know they are going to overwrite the resulting modifiedColumnSet anyways.

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java [hunk 1]
```diff
@@ -219,7 +219,7 @@ private void onUpdate(final TableUpdate upstream) {
                 return;
             }
 
-            final TableUpdateImpl downstream = TableUpdateImpl.copy(upstream);
+            final TableUpdateImpl downstream = TableUpdateImpl.copy(upstream); // TODO: can this be acquire?
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/engine/table/impl/TableUpdateValidator.java:222]: Yes.
```suggestion
            final TableUpdate downstream = upstream.acquire();
```

File: server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java [hunk 2]
```diff
@@ -264,7 +264,7 @@ private Delta(final long step, final long deltaColumnOffset,
                 final BitSet subscribedColumns, final BitSet modifiedColumns) {
             this.step = step;
             this.deltaColumnOffset = deltaColumnOffset;
-            this.update = TableUpdateImpl.copy(update);
+            this.update = TableUpdateImpl.copy(update); // TODO: can this be acquire()
```
Comment (rcaudy) [inline server/src/main/java/io/deephaven/server/barrage/BarrageMessageProducer.java:267]: No. The delta is queued and processed asynchronously w.r.t. the update graph's cycle. MCS are often re-used, and so you need to ensure that you have your own copy of the MCS.
```suggestion
            this.update = TableUpdateImpl.copy(update);
```

**Review by rcaudy — COMMENTED** [review]
I assume you audited all usages of `TableUpdateImpl.copy()`?

**Comment by devinrsmith** [comment]
> I assume you audited all usages of `TableUpdateImpl.copy()`?

Yep, BarrageMessageProducer is the only caller now :)

Kicking off nightlies.

**Pull request merged**